### PR TITLE
Add fallback to `CreateSwapChainForHwnd`

### PR DESCRIPTION
**Pull request by MatkovIvan**
It adds a fallback to the old initialization method in case of any problems.

### CODE REVIEW

File: skiko/src/awtMain/cpp/windows/directXRedrawer.cc [hunk 1]
```diff
@@ -78,18 +94,37 @@ class DirectXDevice
...
+    HRESULT CreateSwapChainForHwnd(IDXGIFactory4 *swapChainFactory4, UINT width, UINT height, IDXGISwapChain1 **swapChain1) {
+        DXGI_SWAP_CHAIN_DESC1 swapChainDesc = {};
+        swapChainDesc.Width = width;
+        swapChainDesc.Height = height;
+        swapChainDesc.Format = DXGI_FORMAT_R8G8B8A8_UNORM;
+        swapChainDesc.SampleDesc.Count = 1;
+        swapChainDesc.SampleDesc.Quality = 0;
+        swapChainDesc.BufferUsage = DXGI_USAGE_RENDER_TARGET_OUTPUT;
+        swapChainDesc.BufferCount = BuffersCount;
+        swapChainDesc.Scaling = DXGI_SCALING_NONE;
```
[the remark this answers is not in the document]
Comment (MatkovIvan) [reply skiko/src/awtMain/cpp/windows/directXRedrawer.cc:125]: It was the value that skiko used before transparency fix

[the remark this answers is not in the document]
Comment (igordmn) [reply skiko/src/awtMain/cpp/windows/directXRedrawer.cc:125]: Should we use `DXGI_SCALING_NONE` there as well?

File: skiko/src/awtMain/cpp/windows/directXRedrawer.cc [hunk 2]
```diff
@@ -60,13 +60,29 @@ class DirectXDevice
...
         UINT height = windowRect.bottom - windowRect.top;
 
         gr_cp<IDXGIFactory4> swapChainFactory4;
         gr_cp<IDXGISwapChain1> swapChain1;
         CreateDXGIFactory2(0, IID_PPV_ARGS(&swapChainFactory4));
+        HRESULT result = CreateSwapChainForComposition(swapChainFactory4.get(), width, height, &swapChain1);
+        if (FAILED(result)) {
+            /*
+             * It's just a fallback path that added for compatibility.
+             * In this case transparency won't be supported.
```
[the remark this answers is not in the document]
Comment (MatkovIvan) [reply skiko/src/awtMain/cpp/windows/directXRedrawer.cc:74]: > Previously we had a fallback to OpenGL, and transparency worked.

It wasn't changed

I don't see the reason to use old approach outside of fallback logic

File: skiko/src/awtMain/cpp/windows/directXRedrawer.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (MatkovIvan) [reply skiko/src/awtMain/cpp/windows/directXRedrawer.cc:125]: As far as I remember it doesn't work with alpha-premultiplied mode.

Hm. Actually [docs](https://learn.microsoft.com/en-us/windows/win32/api/dxgi1_2/ne-dxgi1_2-dxgi_scaling) says it shouldn't work here too 😅 

> The `DXGI_SCALING_NONE` value is supported only for flip presentation model swap chains that you create with the `DXGI_SWAP_EFFECT_FLIP_SEQUENTIAL` value.

So changing it here to `DXGI_SCALING_STRETCH` makes more sense

[the remark this answers is not in the document]
Comment (igordmn) [reply skiko/src/awtMain/cpp/windows/directXRedrawer.cc:125]: Okay, let's change it to DXGI_SCALING_STRETCH

**Comment by MatkovIvan** [comment]
The error was because of zero size, so we need to fix that instead

File: skiko/src/awtMain/cpp/windows/directXRedrawer.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (MatkovIvan) [reply skiko/src/awtMain/cpp/windows/directXRedrawer.cc]: Done

**Review by igordmn — APPROVED** [review]

**Pull request merged**